pkg/client/client.go

```diff
@@ -261,3 +271,22 @@ func MergeAnnotations(toolchainObject client.Object, newAnnotations map[string]s
 	}
 	toolchainObject.SetAnnotations(annotations)
 }
+
+// ApplyUnstructuredObjects applies the given Unstructured objects on the cluster.
+func ApplyUnstructuredObjects(ctx context.Context, cl client.Client, unstructuredObjects []*unstructured.Unstructured, newLabels map[string]string) (bool, error) {
+	applyClient := NewApplyClient(cl)
+	anyApplied := false
+
+	for _, unstructuredObj := range unstructuredObjects {
+		var object = unstructuredObj.DeepCopyObject().(client.Object)
```

Comment:
why do we need to deep-copy the object? this is already done in the `applyObject` function.
Apart from that `*unstructured.Unstructured` is of the type `client.Object` so you can pass it directly to the `ApplyObject` function.

Reply:
Good catch! removed in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/f95ba17b06d5cb6ec1673f3cda2c33b6fb5bf220 . 

Probably a left-over of some old version of the code.